ConsoleCommands/BaseListCommand.cs

```diff
@@ -0,0 +1,144 @@
+﻿using EFT.InventoryLogic;
+using EFT.Trainer.Extensions;
+using EFT.Trainer.Features;
+using JsonType;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System;
+using System.Linq;
+using Comfort.Common;
+using EFT.Interactive;
+
+#nullable enable
+
+namespace EFT.Trainer.ConsoleCommands;
+
+internal abstract class BaseListCommand : LootItemsRelatedCommand
+{
+	public override string Pattern => OptionalArgumentPattern;
+	protected virtual ELootRarity? Rarity => null;
+
+	public override void Execute(Match match)
+	{
+		ListLootItems(match, Rarity);
+	}
+
+	private void ListLootItems(Match match, ELootRarity? rarityFilter = null)
+	{
+		var search = string.Empty;
+		var matchGroup = match.Groups[ValueGroup];
+		if (matchGroup is {Success: true})
+		{
+			search = matchGroup.Value.Trim();
+			if (search == TrackedItem.MatchAll)
+				search = string.Empty;
+		}
+
+		var world = Singleton<GameWorld>.Instance;
+		if (world == null)
+			return;
+
+		var itemsPerName = new Dictionary<string, List<Item>>();
+
+		// Step 1 - look outside containers and inside corpses (loot items)
+		FindLootItems(world, itemsPerName);
+
+		// Step 2 - look inside containers (items)
+		if (LootItems.SearchInsideContainers)
+			FindItemsInContainers(world, itemsPerName);
+
+		var names = itemsPerName.Keys.ToList();
+		names.Sort();
+		names.Reverse();
+
+		var count = 0;
+		foreach (var itemName in names)
+		{
+			if (itemName.IndexOf(search, StringComparison.OrdinalIgnoreCase) < 0)
+				continue;
+
+			var list = itemsPerName[itemName];
+			var rarity = list.First().Template.GetEstimatedRarity();
+			if (rarityFilter.HasValue && rarityFilter.Value != rarity)
+				continue;
+
+			var extra = rarity != ELootRarity.Not_exist ? $" ({rarity.Color()})" : string.Empty;
+			AddConsoleLog($"{itemName} [{list.Count.ToString().Cyan()}]{extra}");
+
+			count += list.Count;
+		}
+
+		AddConsoleLog("------");
+		AddConsoleLog($"found {count.ToString().Cyan()} item(s)");
```

Comment:
Would it be possible to list where we found this item? i.e. like inside a container, scav, pmc etc?